ui/src/AppCharting/AppChartC3.svelte

```diff
@@ -258,15 +251,22 @@
   }
 
   function styleSpecialPoints() {
-    styleDataPointErrors();
-    styleMissingPoints();
+    // Microtask Hack.
+    setTimeout(() => {
+      styleDataPointErrors();
+      styleMissingPoints();
+    }, 0);
   }
+
+  $: redrawKey = chartType || chartData || axis || Date.now();
 </script>
 
-<LineChart
-  {axis}
-  data={chartData}
-  height={DEFAULT_CHART_HEIGHT}
-  {legendTitle}
-  on:rendered={styleSpecialPoints}
-/>
+{#key redrawKey}
```

Comment:
I think the chart was already always updating correctly.

Reply:
The problem wasn't the chart updating, the problem was that changed charts were still emitting their `onrender` event even after the chart was removed from the dom.   This causes the svelte `onDestroy()` function to get called cleaning up and making sure `chartRef.destroy()` is called.  

It seemed like a super niche C3 issue.